presto-main/src/main/java/com/facebook/presto/execution/SqlMaterializedQueryTableRefresher.java

```diff
@@ -0,0 +1,234 @@
+            Expression expression = sqlParser.createExpression(predicateForMaterializedQueryTable);
+            changesToMaterializedQueryTable = Optional.of(expression);
+        }
+
+        // materializedQueryTable is always the fully qualified name.
+        Delete delete = new Delete(new Table(materializedQueryTableName), changesToMaterializedQueryTable);
+        QueryId deleteQueryId = queryIdGenerator.createNextQueryId();
+        QueryInfo queryInfo = queryManager.createQuery(session, SqlFormatter.formatSql(delete), deleteQueryId);
+        queryInfo = waitForQueryToFinish(queryInfo, deleteQueryId);
+        if (queryInfo.getState() != FINISHED) {
+            //            transactionManager.asyncAbort(transactionId);
+            throw new PrestoException(REFRESH_TABLE_FAILED, String.format("Failed to delete from materialized query table %s", materializedQueryTable));
+        }
+
+        Expression refreshPredicateForBaseTables = parseBaseTablePredicates(predicateForBaseTables);
+        if (!BooleanLiteral.TRUE_LITERAL.equals(refreshPredicateForBaseTables)) {
+            QuerySpecification oldQuerySpecification = (QuerySpecification) materializedQuery.getQueryBody();
+            Optional<Expression> originalPredicateToBaseTable = oldQuerySpecification.getWhere();
+            Expression predicateForBaseTable = originalPredicateToBaseTable.isPresent() ? new LogicalBinaryExpression(LogicalBinaryExpression.Type.AND, originalPredicateToBaseTable.get(), refreshPredicateForBaseTables) : refreshPredicateForBaseTables;
+
+            QueryBody newQuerySpecification = new QuerySpecification(
+                    oldQuerySpecification.getLocation(),
+                    oldQuerySpecification.getSelect(),
+                    oldQuerySpecification.getFrom(),
+                    Optional.of(predicateForBaseTable),
+                    oldQuerySpecification.getGroupBy(),
+                    oldQuerySpecification.getHaving(),
+                    oldQuerySpecification.getOrderBy(),
+                    oldQuerySpecification.getLimit());
+
+            materializedQuery = new Query(
+                    materializedQuery.getWith(),
+                    newQuerySpecification,
+                    materializedQuery.getOrderBy(),
+                    materializedQuery.getLimit(),
+                    materializedQuery.getApproximate());
+        }
+
+        Insert insert = new Insert(materializedQueryTableName, Optional.empty(), materializedQuery);
+        QueryInfo insertQueryInfo = queryManager.createQuery(session, SqlFormatter.formatSql(insert), session.getQueryId());
+        insertQueryInfo = waitForQueryToFinish(insertQueryInfo, session.getQueryId());
+
+        if (insertQueryInfo.getState() != FINISHED) {
+            //            transactionManager.asyncAbort(transactionId);
+            throw new PrestoException(REFRESH_TABLE_FAILED, String.format("Failed to insert into materialized query table %s", materializedQueryTable));
```

Comment:
Should I keep it as             
throw new PrestoException(REFRESH_TABLE_FAILED, format("Failed to insert into materialized query table '%s'", materializedQueryTable));
since it is better to keep single quote around the materializedQueryTable name?